ikarus/finiteelements/mechanics/materials/muesli/mueslihelpers.hh

```diff
@@ -0,0 +1,132 @@
+// SPDX-FileCopyrightText: 2021-2025 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#pragma once
+
+#if ENABLE_MUESLI
+
+  #include <muesli/muesli.h>
+
+  #include <ikarus/finiteelements/physicshelper.hh>
+  #include <ikarus/utils/tensorutils.hh>
+
+namespace Ikarus::Concepts {
+/**
+ * \concept MuesliMaterialImplementation
+ * \brief Concept to check weather a type is a Mueli material implementation
+ *
+ * \tparam MAT the material type
+ */
+template <typename MAT>
+concept MuesliMaterialImplementation = requires { std::is_base_of_v<muesli::material, MAT>; };
+} // namespace Ikarus::Concepts
+
+namespace Ikarus::Materials {
+
+/**
+ * \brief converts Ikarus material parameters to Muesli material properties.
+ *
+ * \tparam MPT the type of the Ikarus material parameters.
+ * \param mpt the Ikarus material parameters.
+ * \return MaterialProperties muesli material properties with the Lambda and mu set
+ */
+template <Concepts::MPTuple MPT>
+inline muesli::materialProperties propertiesFromIkarusMaterialParameters(const MPT& mpt) {
+  auto converter = convertLameConstants(mpt);
+
+  auto mpm = muesli::materialProperties{};
+  mpm.insert({"lambda", converter.toLamesFirstParameter()});
+  mpm.insert({"mu", converter.toShearModulus()});
+
+  return mpm;
+}
+
+/**
+ * \brief adds a specific tag to the muesli materialproperties, thic can be used for example to add the `regularized`
+ * tag for Neo-Hooke or the `compressible` tag for Yeoh and Arruda-Boyce.
+ */
+inline void addTag(muesli::materialProperties& mpm, const std::string& tagName, double tagValue = 0) {
+  mpm.insert({tagName, tagValue});
+}
+
+/**
+ * \brief Converts the entries of a Eigen::Matrix to a provided muesli::istensor (symmetric 2nd order tensor).
+ *
+ * \param it provided istensor.
+ * \param C the Eigen::Matrix that is to be converted.
+ */
+inline void toistensor(istensor& it, const Eigen::Matrix<double, 3, 3>& C) {
```

Comment:
since istensor is a fixed size object i would suggest return istensor it with return instead of input-output argument by ref.